mars/services/storage/core.py

```diff
@@ -554,24 +553,65 @@ async def open_writer(self,
         return WrappedStorageFileObject(writer, level, size, session_id, data_key,
                                         self._data_manager_ref, self._clients[level])
 
-    async def list(self, level: StorageLevel) -> List:
-        return await self._clients[level].list()
+    async def _get_meta_api(self, session_id: str):
+        if self._supervisor_address is None:
+            cluster_api = await ClusterAPI.create(self.address)
+            self._supervisor_address = (await cluster_api.get_supervisors())[0]
+        return await MetaAPI.create(session_id=session_id,
+                                    address=self._supervisor_address)
 
     async def fetch(self,
                     session_id: str,
-                    data_key: str):
-        if StorageLevel.REMOTE not in self._clients:
-            raise NotImplementedError
-        else:
-            data_info = await self._data_manager_ref.get_data_info(
-                session_id, data_key)
-            await self._clients[StorageLevel.REMOTE].fetch(data_info.object_id)
+                    data_key: str,
+                    level: StorageLevel,
+                    address: str,
+                    band_name: str,
+                    error: str):
+        from .transfer import SenderManagerActor
```

Comment:
Better to separate into multiple functions, like `fetch_local`, `fetch_remote`, `fetch_via_transfer`.